### PR TITLE
Update CHANGELOG for v2.36

### PR DESCRIPTION
**Pull request by metalmatze**
I've compiled the list of changes and updated our CHANGELOG.md accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+* [ENHANCEMENT] Changing TotalQueryableSamples from int to int64. #10549
+* [ENHANCEMENT] Move from `io/ioutil` to `io` and `os` packages. #10528
+* [ENHANCEMENT] Add stripDomain to template function. #10475
+* [ENHANCEMENT] Send target and metadata cache in context (again). #10636
+* [ENHANCEMENT] Enable active search through dropped targets. #10668
+* [ENHANCEMENT] Stop rule manager before TSDB is stopped. #10680
+* [ENHANCEMENT] promtool: support matchers when querying label values. #10727
+* [BUGFIX] Pass the correct context to staleness Appender. #10588
+* [BUGFIX] tsdb/agent: Ignore duplicate exemplars. #10595
+* [BUGFIX] Fix chunk overflow appending samples at a variable rate. #10607
```
Comment (roidelapluie) [inline CHANGELOG.md:21]:
```suggestion
* [BUGFIX] TSDB: Fix chunk overflow appending samples at a variable rate. #10607
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
 # Changelog
 
+## 2.36.0-rc0 / 2022-05-24
+
+This Prometheus release uses go 1.18 and needs at least go1.17 going forward.
+
+* [FEATURE] Add lowercase and uppercase relabel action. #10641
+* [FEATURE] Add hidden flag to configure discovery loop interval. #10634
+* [FEATURE] Add Linode SD failure count metric. #10673
+* [FEATURE] Add service discovery for IONOS Cloud. #10514
```
Comment (roidelapluie) [inline CHANGELOG.md:10]:
```suggestion
* [FEATURE] SD: Add service discovery for IONOS Cloud. #10514
```

**Review by roidelapluie — COMMENTED** [review]
Awesome 👍

**Comment by metalmatze** [comment]
Nice! I'll update these. 
The intention was to give more updates so that we can easily delete unnecessary ones. :+1:

**Comment by metalmatze** [comment]
Great VSCode auto formatted the _entire_ CHANGELOG.md. Fixing.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+This Prometheus release uses go 1.18 and needs at least go1.17 going forward.
+
+* [FEATURE] Add lowercase and uppercase relabel action. #10641
+* [FEATURE] SD: Add IONOS Cloud integration. #10514
+* [FEATURE] SD: Add Vultr integration. #10714
+* [FEATURE] SD: Add Linode SD failure count metric. #10673
+* [FEATURE] Add prometheus_ready metric. #10682
+* [ENHANCEMENT] Add stripDomain to template function. #10475
+* [ENHANCEMENT] UI: Enable active search through dropped targets. #10668
+* [ENHANCEMENT] Stop rule manager before TSDB is stopped. #10680
```
Comment (prymitive) [inline CHANGELOG.md:14]: IMHO this is more of a `BUGFIX` for me since previous releases would trigger false positive alerts

Comment (metalmatze) [reply CHANGELOG.md:14]: Updated. :+1:

**Pull request merged**